overlord/ifacestate/ifacemgr.go

```diff
@@ -77,6 +77,54 @@ func Manager(s *state.State, hookManager *hookstate.HookManager, extraInterfaces
 	return m, nil
 }
 
+func setInitialConnectAttributes(ts *state.Task, plugSnap string, plugName string, slotSnap string, slotName string) error {
+	// Set initial interface attributes for the plug- and slot- snaps in connect task.
+	var snapst snapstate.SnapState
+	var err error
+
+	st := ts.State()
+	if err = snapstate.Get(st, plugSnap, &snapst); err != nil {
+		return err
+	}
+
+	snapInfo, err := snapst.CurrentInfo()
+	if err != nil {
+		return err
+	}
+
+	plugAttrs := make(map[string]interface{})
+	if plug, ok := snapInfo.Plugs[plugName]; ok {
+		for k, v := range plug.Attrs {
+			plugAttrs[k] = v
+		}
+	} else {
+		return fmt.Errorf("Snap %q has no plug named %q", plugSnap, plugName)
+	}
+
+	if err = snapstate.Get(st, slotSnap, &snapst); err != nil {
+		return err
+	}
+	snapInfo, err = snapst.CurrentInfo()
+	if err != nil {
+		return err
+	}
+	snap.AddImplicitSlots(snapInfo)
```

Comment:
setInitialConnectAttributes expects plug and slot to be present in the snapInfo, so that it can copy their attributes into "plug-attrs" and "slot-attrs". If the slot (or plug) is missing, it will return "snap %q has no slot named %q". AddImplicitSlots is needed to augment slotInfo with the slots of core snap, otherwise we fail here on every 'connect' concerning core snap.

I'm not sure there is a better place to do this, it's only needed in setInitialConnectAttributes at this point, it doesn't make sense to move it outside to Connect(..) for example.